cmake/CheckTrezor.cmake

```diff
@@ -75,19 +75,25 @@ if (USE_DEVICE_TREZOR)
     # Protobuf is required to build protobuf messages for Trezor
     include(FindProtobuf OPTIONAL)
 
-    FIND_PACKAGE(Protobuf CONFIG)
+    # PkgConfig works better with new Protobuf
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(PROTOBUF protobuf)
+
+    if (NOT Protobuf_FOUND)
+        FIND_PACKAGE(Protobuf CONFIG)
+    endif()
     if (NOT Protobuf_FOUND)
         FIND_PACKAGE(Protobuf)
     endif()
 
     _trezor_protobuf_fix_vars()
 
     # Early fail for optional Trezor support
-    if(${Protobuf_VERSION} GREATER 21)
-        trezor_fatal_msg("Trezor: Unsupported Protobuf version ${Protobuf_VERSION}. Please, use Protobuf v21.")
+    if(${CMAKE_CXX_STANDARD} LESS 17 AND ${Protobuf_VERSION} GREATER 21)
```

Comment:
If there is no protobuf version installed, this `if` fails with this CMake error

```
CMake Error at cmake/CheckTrezor.cmake:92 (if):
  if given arguments:

    "17" "LESS" "17" "AND" "GREATER" "21"

  Unknown arguments specified
Call Stack (most recent call first):
  CMakeLists.txt:704 (include)


-- Configuring incomplete, errors occurred!
```

Which aborts CMake. This is unintended behaviour for `USE_DEVICE_TREZOR=ON` and `USE_DEVICE_TREZOR_MANDATORY=OFF`. The issue already exists in the current code (see https://github.com/monero-project/monero/issues/9040) but it can be fixed together with this PR.

Reply:
aah you are right, I exchanged those two ifs so first protobuf is checked. it fixes it.